### DISCUSSION TITLE
关于脚注编号 & 脚注的排版

### DISCUSSION THREAD

**Discussion by dreistein543** [nju-lug/NJUThesis#127] [Ideas]
目前，参考说明文档，使用 `\footnote` 来添加脚注，但是在编号上存在两个问题；

1. 目前的编号是简单的阿拉伯数字编号，但通常我门都使用上标圆圈数字 ① 来编号
2. 目前的脚注是跨页连续编号的，但通常我们使用每一页重新编号

希望能加上设置选项让我们自由选择编号方式

另一个是脚注的排版，目前默认是首行缩进1.5字符，例如：
<img width="514" alt="截屏2022-04-14 21 57 39" src="https://user-images.githubusercontent.com/30745407/163406254-ccad1828-a981-4645-a506-f4701a766746.png">

但是从美学以及方便读者快速识别不同脚注的情况考虑，应该使用悬挂缩进 1 字符（即类似于参考文献 numeric 的样式）

**Comment by atxy-blip** [comment]
> 使用上标圆圈数字 ① 来编号
> 每一页重新编号

隔壁 `fduthesis` 实现效果如下图，打算直接拿来抄抄。

![image](https://user-images.githubusercontent.com/63113546/163511807-754d35c6-6268-4f57-8558-2a96ea56a8bc.png)

![image](https://user-images.githubusercontent.com/63113546/163511569-56a4244f-9e1a-429f-b2fc-d09e54d28eac.png)

> 悬挂缩进 1 字符

悬挂缩进实际上是 `footmisc` 包的一个选项，我再看看怎么提供选择吧。

```TeX
\usepackage[hang]{footmisc} 
```

**Reply by atxy-blip** [reply]
v0.18.0 可以使用带圈数字。

**Reply by dreistein543** [reply]
感恩，五一节快乐

**Reply by dreistein543** [reply]
发现在 v18.0 的模板中，`\usepackage[hang]{footmisc} ` 会报错 显示 `LaTeX Error: Option clash for package footmisc.`

**Reply by atxy-blip** [reply]
为了增加带圈数字重新定义了部分相关的宏，可能引发了这个兼容性问题。我也注意到 `\PassOptionsToPackage` 似乎没对 `footmisc` 正确生效。下次再整整。

**Reply by vill-jiang** [reply]
在 v18.0 的使用 `footer = { pifont },` 时出现错误 
```Package fancyhdr Error: Illegal char `p' in \fancyfoot argument: [p].```
![image](https://user-images.githubusercontent.com/24198247/166445423-11418bc5-a048-421c-91cd-0d04345c1196.png)

**Reply by atxy-blip** [reply]
`footer` 是用于设置页脚内容的，你需要的是 `footnote-style` 设置项：
![image](https://user-images.githubusercontent.com/63113546/166446077-6614d17f-ed55-430c-8ab0-dff166728c61.png)

**Reply by vill-jiang** [reply]
照着 example.tex 改的，大意了。😂
谢谢。
